src/commands/config/ConfigDungeons.ts

```diff
@@ -1297,6 +1309,15 @@ export class ConfigDungeons extends BaseCommand {
                     cDungeon.allowedModifiers = res;
                     break;
                 }
+                case "require_loc": {
+                    const newStyle = !cDungeon.locationToProgress ? "SUCCESS" : "DANGER";
+                    configLocationRequirement.setStyle(newStyle);
+
+                    await botMsg.edit({ embeds: [embed], components: AdvancedCollector.getActionRowsFromComponents(buttons) });
```

Comment:
I don't think you need to call the `edit` method here. 

What you can do is, at the beginning of the loop, set the style of the button there as opposed to here. Then, when someone presses the "Require Location" button, all you need to do here is change the value associated w/ the property `locationToProgress` to the desired value and then let the loop handle editing the button's style for you. 

See how I changed `saveButton`'s disabled status at the very beginning of the loop.